### PR TITLE
add check of target before using wasmer as sandbox executor

### PR DESCRIPTION
**Pull request by grishasobol**

### CODE REVIEW

File: client/executor/common/src/sandbox.rs [hunk 1]
```diff
@@ -430,6 +430,25 @@ enum BackendContext {
 	Wasmer(WasmerBackend),
 }
 
+#[cfg(feature = "wasmer-sandbox")]
+fn target_suit_for_wasmer() -> bool {
```
Comment (shamilsan) [inline client/executor/common/src/sandbox.rs:434]:
```suggestion
fn is_target_wasmer_compatible() -> bool {
```

File: client/executor/common/src/sandbox.rs [hunk 2]
```diff
@@ -439,8 +458,22 @@ impl BackendContext {
...
+					BackendContext::Wasmi
+				}
+			},
+
+			#[cfg(feature = "wasmer-sandbox")]
+			SandboxBackend::Wasmer => {
+				if target_suit_for_wasmer() {
+					BackendContext::Wasmer(WasmerBackend::new())
+				} else {
+					panic!("Cannot use wasmer in this target")
```
Comment (shamilsan) [inline client/executor/common/src/sandbox.rs:474]:
```suggestion
					panic!("CPU architechture isn't compatible with `wasmer`")
```

File: client/executor/common/src/sandbox.rs [hunk 3]
```diff
@@ -430,6 +430,25 @@ enum BackendContext {
...
+			if target.cpu_features().contains(CpuFeature::AVX) {
+				true
+			} else if target.cpu_features().contains(CpuFeature::SSE42) {
+				true
+			} else {
+				false
+			}
+		}
+		Architecture::Aarch64(_) => true,
+		_ => false,
```
[the remark this answers is not in the document]
Comment (grishasobol) [reply client/executor/common/src/sandbox.rs:448]: No, there is no support from wasmer there

File: client/executor/common/src/sandbox.rs (hunk 2 above)
Comment (grishasobol) [reply client/executor/common/src/sandbox.rs:474]: Done

File: client/executor/common/src/sandbox.rs [hunk 4]
```diff
@@ -430,6 +430,19 @@ enum BackendContext {
 	Wasmer(WasmerBackend),
 }
 
+#[cfg(feature = "wasmer-sandbox")]
+fn is_target_suit_for_wasmer() -> bool {
```
Comment (shamilsan) [inline client/executor/common/src/sandbox.rs:434]: I don't like `suit` here, sorry 😄
`is_suitable` or `does_suit` is more grammatically correct.

Comment (grishasobol) [reply client/executor/common/src/sandbox.rs:434]: Done

**Pull request merged**